### PR TITLE
FAD Add release notes to update dialog

### PR DESCRIPTION
**Pull request by andrewbibiloni**
- Add release notes to update dialog if available 
- add tests for when release notes are empty 
- nit: change to more accurate test names for signInTester

### CODE REVIEW

File: firebase-app-distribution/src/test/java/com/google/firebase/appdistribution/FirebaseAppDistributionTest.java [hunk 1]
```diff
@@ -77,6 +77,15 @@
           .setDownloadUrl("https://test-url")
           .build();
 
+  private static final AppDistributionReleaseInternal TEST_RELEASE_NEWER_AAB_INTERNAL_2 =
```
Comment (rachaprince) [inline firebase-app-distribution/src/test/java/com/google/firebase/appdistribution/FirebaseAppDistributionTest.java:80]: I don't think we need to create a second object for testing. If we turn these into builder objects (AppDistributionReleaseInternal.Builder), then the tests can write:

testReleaseNewerAabInternal.build() // no release notes
testReleaseNewerAaabInternal.setReleaseNotes(....).build() // with release notes

This blog has a good explanation of the design pattern: https://testing.googleblog.com/2018/02/testing-on-toilet-cleanly-create-test.html 

(If you adopt this pattern, we may want to change these private fields into private methods that return a builder when called. )

File: firebase-app-distribution/src/main/java/com/google/firebase/appdistribution/FirebaseAppDistribution.java [hunk 2]
```diff
@@ -383,11 +383,22 @@ private void showUpdateAlertDialog(AppDistributionRelease latestRelease) {
     AlertDialog alertDialog = new AlertDialog.Builder(currentActivity).create();
     alertDialog.setTitle(context.getString(R.string.update_dialog_title));
 
-    alertDialog.setMessage(
-        String.format(
-            "Version %s (%s) is available.",
-            latestRelease.getDisplayVersion(), latestRelease.getBuildVersion()));
+    String message;
+    if (latestRelease.getReleaseNotes().equals("")) {
```
[the remark this answers is not in the document]
Comment (andrewbibiloni) [reply firebase-app-distribution/src/main/java/com/google/firebase/appdistribution/FirebaseAppDistribution.java:387]: Pranav added a method called tryGetValue() in the client that sets fields to empty string if retrieval fails so I don't think it can be null anymore, but I can add a check to be defensive

**Review by rachaprince — APPROVED** [review]
